lms/djangoapps/commerce/api/v1/views.py

```diff
@@ -64,10 +73,17 @@ class OrderView(APIView):
     """ Retrieve order details. """
 
     authentication_classes = (SessionAuthentication,)
-    permission_classes = (IsAuthenticated,)
+    permission_classes = (IsAuthenticatedOrActivationOverridden,)
```

Comment:
Alright, I'll leave it to you to make the final decision on how to proceed. If we do keep this code, perhaps you can find a way to refactor the common lines of code in the permission class and this view to keep things DRY.